### DISCUSSION TITLE
No Permission / Urgente

### DISCUSSION THREAD

**Discussion by SupervisoreOP** [fattureincloud/api#259] [Q&A] [answered]
Buongiorno,

fino ad oggi ho testato correttamente le chiamate V2 (crea fattura issued_documents) tramite un account creato appositamente per eseguire i test e tutto funzionava alla perfezione. 
Oggi invece che dovevo far partire le call V2 ufficialmente e verso il mio account ufficiale non funziona perché mi risponde:
  "error": {
    "message": "No permission.",
    "code": "NO_PERMISSION"

eppure tutto è identico ai test eseguiti sull'account di prova!
Ho controllato che il token sia corretto, ho controllato l'applicazione collegata che è corretta, ho provato a scollegare e ricollegare ridando tutti i permessi manuali e completi (premetto che sto utilizzando il token manuale) ma niente l'errore è sempre NO PERMISSION.

Io ho urgente bisogno di far funzionare la stessa identica chiamata che verso altro account funziona!! E' questo che non mi spiego, se cambio token (di altro account) funziona.

Potete aiutarmi?
Il codice è corretto, proprio perché è lo stesso utilizzato per i test. Inoltre ho anche provato il codice di test che c'è sulle vostre API reference l'errore è lo stesso...

Perché? ho dato gli accessi completi a tutti i moduli.

Grazie

**Comment by SupervisoreOP** [comment]
Aggiungo che l'ambito di intervento era semplice: Cambiare l'ID company e il TOKER per l'autorizzazione. Mi aspettavo fosse una gioco da ragazzi invece no...Nonostante ho settato tutto in modo identico sulla Piattaforma Fatture in Cloud.

Aggiungo anche che il problema sembra presentarsi solo sulla creazione Fatture, ho provato in modo sciocco a creare un Payment account tramite API, stesso token e funziona...rilevate qualche problema?

Attendo vostre, grazie

**Comment by emajo** [comment]
Ciao @SupervisoreOP, lato nostro non stiamo avendo problemi di questo genere, 401 lo si ottiene proprio per un errato token/company id, la company su cui sta operando per la creazione della fattura è la medesima che per la creazione del payment account?

Grazie,
Emanuele.

**Reply by SupervisoreOP** [reply]
> Ciao @SupervisoreOP, lato nostro non stiamo avendo problemi di questo genere, 401 lo si ottiene proprio per un errato token/company id, la company su cui sta operando per la creazione della fattura è la medesima che per la creazione del payment account?
> 
> Grazie, Emanuele.

Buongiorno, ho appena riprovato. Ho recuperato l'id Company tramite la call LIST USER COMPANY, quindi il token è valido.
Recupero l'id company e lo uso insieme al token per fare questa call:

`{
"data": {

"type": "invoice",

"amount_net": 0,

"amount_due_discount": 0,
"entity": {
"id": 0,
"name": “FULL NAME”,
"tax_code": “CF”,
"address_street": “INDIRIZZO”,
"address_postal_code": “CAP”,
"address_city": “CITTA”,
"address_province": “PROV”,
"address_extra": "Extra”,
"country": "Italia",
"email": "",
"certified_email": “”
},
"numeration": "",
"subject": "Soggetto",
"visible_subject": "Ordine #”,
"notes": "Copia del documento elettronico disponibile nella Sua area riservata dell'Agenzia delle entrate.",
"currency": {
"id": "EUR",
"exchange_rate": "1.00000",
"symbol": "€"
},
"language": {
"code": "it",
"name": "Italiano"
},
"items_list": [
{
"product_id": 0,
"code": "PRD1",
"name": "Riparazione & Supporto",
"net_price": 0,
"gross_price": 100,
"category": "Meccanica",
"discount": 0,
"qty": 1,
"not_taxable": false,
"vat": {
"id": 0
}
}
],
"payments_list": [
{
"amount": 100,
"due_date": “2023-02-02”,
"paid_date": “2023-02-02”,
"status": "paid",
"payment_account": {
"id": 797248
}
}
],
"payment_method":{
"name": "SumUp"
},
"use_gross_prices": true,
"show_payment_method": true,
"show_payments": false
}
}`

Ma nulla  non funziona....Io ho urgente bisogno...Ho collegato/scollegato, dato e ridato le grant...ma nulla

Ho anche recuperato nuovamente il PAYMENT ACCOUNT tramite l'id Company...( List Payment Accounts)...ma nulla

**Reply by emajo** [reply]
Ti consiglio di scrivere al supporto clienti così da poter guardare i log delle chiamate.

**Reply by SupervisoreOP** [reply]
> Ti consiglio di scrivere al supporto clienti così da poter guardare i log delle chiamate.

Scusate per non allungare ulteriormente i tempi, se vi do qui tutte le info sulle chiamate (che vedo da mio LOG) non potete vedere i dettagli voi? Ripeto che siamo fermi con la creazione fatture....
Aggiungo che ho provato semplicemente una call LIST ISSUED DOCUMENT con ID company e TOKEN e mi dice NO PERMISSION Ugualmente....non capisco quale sia il problema

Ho contattato il supporto utenti ma immagino non sia una cosa velocissima

**Comment by SupervisoreOP** [comment, marked as answer]
Mi sa cho ho capito dove è il problema (LIST USER COMPANY):

`{
  "data": {
    "companies": [
      {
        "id": XXXXX,
        "name": "XXXXX",
        "alias": null,
        "vat_number": "XXXX",
        "tax_code": "XXXX",
        "type": "company",
        "connection_id": XXXXX,
        "controlled_companies": [],
        "fic": true,
        "dic": false,
        "fic_plan": "premium_plus",
        "fic_license_expire": "2023-12-20",
        "permissions": {
          "fic_situation": "none",
          "fic_clients": "none",
          "fic_suppliers": "none",
          "fic_products": "write",
          "fic_issued_documents": "none",
          "fic_received_documents": "none",
          "fic_receipts": "none",
          "fic_calendar": "none",
          "fic_archive": "none",
          "fic_taxes": "none",
          "fic_stock": "none",
          "fic_cashbook": "none",
          "fic_settings": "write",
          "fic_emails": "none",
          "dic_employees": "none",
          "dic_timesheet": "none",
          "dic_settings": "none"
        }
      }
    ]
  }
}`

Vedo che         "fic_issued_documents": "none", è NONE. ma io ho dato le Grant quando ho collegato l'azienda all'applicazione...sembra non averle prese. Come le posso modificare in WRITE?

Grazie

**Reply by emajo** [reply]
Innanzitutto, stiamo parlando di token manuale o oauth?

**Reply by SupervisoreOP** [reply]
> Innanzitutto, stiamo parlando di token manuale o oauth?

Token Manuale. a cui ho dato tutte le grant quando ho eseguito il collegamento dell'applicazione all'azienda, ma sembra non averle prese.

E' molto strano perche abbiamo eseguito le stesse identiche call, su un altro account di test (ovviamente diverso ID company e token) e tutto funzionava, abbiamo solo cambiato i parametri essenziali: Token e Id company...ma tutto è identico

**Reply by emajo** [reply]
Andando in applicazioni collegate, ti risulta che i moduli a cui ha accesso l'app siano quelli corretti?

**Reply by SupervisoreOP** [reply]
> Andando in applicazioni collegate, ti risulta che i moduli a cui ha accesso l'app siano quelli corretti?

Si ho dato la grant a tutti i moduli tra cui Fatture Accesso completo, anche Fatture Ricevute accesso completo...Perchè invece in Permission vedo Fic_issued_documents": "none" ?

<img width="592" alt="Schermata 2023-02-02 alle 09 24 54" src="https://user-images.githubusercontent.com/122611532/216273589-8d2be3e5-1685-4c98-8614-917dd7eaf9fd.png">

**Reply by valmoz** [reply]
Buongiorno @SupervisoreOP 
per escludere che sia il token o la company a creare problemi, le consiglio di provare in parallelo anche da API Reference. In questo modo potremo capire se effettivamente c'è un bug sul token o se dobbiamo concentrarci sul codice.

Altra possibilità, nel caso in cui il test precedente su API Reference dovesse fallire, è quella di provare a creare una nuova app e/o un nuovo token.

Abbiamo allertato anche il supporto clienti, purtroppo nel loro caso hanno delle priorità dettate dal loro software.

A disposizione
Mauro

**Reply by SupervisoreOP** [reply]
> Buongiorno @SupervisoreOP per escludere che sia il token o la company a creare problemi, le consiglio di provare in parallelo anche da API Reference. In questo modo potremo capire se effettivamente c'è un bug sul token o se dobbiamo concentrarci sul codice.
> 
> Altra possibilità, nel caso in cui il test precedente su API Reference dovesse fallire, è quella di provare a creare una nuova app e/o un nuovo token.
> 
> Abbiamo allertato anche il supporto clienti, purtroppo nel loro caso hanno delle priorità dettate dal loro software.
> 
> A disposizione Mauro

Buongiorno,

ho provato anche da API Reference con la vostra call di esempio, ma nulla stesso errore, ecco la call:

`{
  "data": {
    "type": "receipt",
    "numeration": "rec123",
    "subject": "",
    "visible_subject": "",
    "amount_net": 68.18,
    "amount_vat": 6.82,
    "amount_due_discount": 0,
    "entity": {
      "id": 54321,
      "name": "Mary Red S.r.L.",
      "vat_number": "IT05432181211",
      "tax_code": "IT05432181211",
      "address_street": "Via Italia, 66",
      "address_postal_code": "20900",
      "address_city": "Milano",
      "address_province": "MI",
      "address_extra": "",
      "country": "Italia",
      "certified_email": "mary@pec.red.com",
      "ei_code": "ABCXCR1"
    },
    "date": "2021-08-20",
    "number": 1,
    "next_due_date": "2021-12-31",
    "items_list": [
      {
        "product_id": 333,
        "code": "SG3",
        "name": "Soggiorno",
        "measure": "",
        "net_price": 68.18182,
        "category": "",
        "id": 277875565,
        "gross_price": 75,
        "apply_withholding_taxes": true,
        "discount": 0,
        "discount_highlight": false,
        "in_dn": false,
        "qty": 1,
        "vat": {
          "id": 3,
          "value": 10,
          "description": ""
        },
        "stock": true,
        "description": "",
        "not_taxable": false
      }
    ],
    "payments_list": [
      {
        "amount": 75,
        "due_date": "2020-08-23",
        "id": 444,
        "payment_terms": {
          "days": 0,
          "type": "standard"
        },
        "status": "not_paid"
      }
    ],
    "payment_method": {
      "name": "SumUp"
    }
  }
}`

Ho provato a provato ieri e anche stamattinna una nuova APP e collegarla per avere un nuovo Token ma niente ugualmente...sempre NO PERMISSION.

Il problema non è il codice....

Comunque ho un dubbio, il proprietario principale dell'account non sono io, lui ha creato per me il mio secondo account sviluppatore che ha accesso alla sua piattaforma (per eseguire gli adeguamenti alla V2). Io però non ho gli accessi a tutti i moduli sulla piattaforma FIC...però non credo che questo possa influire o sbaglio? altrimenti le GRANT nel collegamento dell'applicazioni non servirebbero a nulla.

Il Token dovrebbe funzionare e non basarsi sulle grant dei moduli che un User ha, giusto?

**Reply by valmoz** [reply]
Buongiorno @SupervisoreOP 
nel dubbio, provi a fare eseguire la procedura di connessione a lui e si faccia passare il token.

**Reply by SupervisoreOP** [reply]
> Buongiorno @SupervisoreOP
> nel dubbio, provi a fare eseguire la procedura di connessione a lui e si faccia passare il token.

Buongiorno, ok farò provare appena possibile. Nel frattempo dovrò attendere il servizio clienti per le verifiche del caso? Perché comunque un problema di base c’è…

**Reply by valmoz** [reply]
purtroppo senza scambiare dettagli come il company id non possiamo verificare dai log... e qui non è possibile scambiare il tipo di informazioni che ci servono, essendo un forum pubblico.

**Reply by valmoz** [reply]
Buongiorno @SupervisoreOP 
purtroppo dal supporto clienti ancora nessuna segnalazione a riguardo. Ha provato a contattarli?

Comunque ho provato a cercare i log relativi alle sue chiamate, e credo di esser riuscito a individuare la company su cui sta operando (non ne ho la certezza, non avendo a disposizione i dati che mi avrebbe fornito il supporto). Vedo però che l'errore segnalato in questa discussione avviene solo per una delle due app presenti sul vostro account, mentre l'altra ha registrato numerosi 200 OK e 422 Invalid Request (errori di validazione), corretto?

Resto in attesa di una sua risposta.
Grazie
Mauro

**Reply by SupervisoreOP** [reply]
> Buongiorno @SupervisoreOP purtroppo dal supporto clienti ancora nessuna segnalazione a riguardo. Ha provato a contattarli?
> 
> Comunque ho provato a cercare i log relativi alle sue chiamate, e credo di esser riuscito a individuare la company su cui sta operando (non ne ho la certezza, non avendo a disposizione i dati che mi avrebbe fornito il supporto). Vedo però che l'errore segnalato in questa discussione avviene solo per una delle due app presenti sul vostro account, mentre l'altra ha registrato numerosi 200 OK e 422 Invalid Request (errori di validazione), corretto?
> 
> Resto in attesa di una sua risposta. Grazie Mauro

Buongiorno @valmoz Ieri ho sfatato il mio dubbio facendomi restituire un nuovo Token creato con una nuova Applicazione e poi collegato all'azienda direttamente dal proprietario dell'account e sono infatti riuscito ad eseguire la chiamata.

Quindi come avevo immaginato il problema erano le grant sul mio account....Ma questo è davvero molto strano e non credo sia corretto.

Sono riuscito ad eseguire gli adeguamenti e ora le prime call si stanno finalizzando. Resto in attesa da parte del proprietario che sta facendo le opportune verifiche in produzione.

Se ci fossero altri problemi collegati al thread, vi farò sapere.

Saluti

**Reply by valmoz** [reply]
Ok, perfetto. Dal canto nostro provvederemo con le verifiche per determinare la causa dell'inconveniente.

Per ora chiudo la conversazione, nel caso dovessero esserci altri problemi collegati rimuoverò lo stato "answered".

Grazie
Mauro
